### PR TITLE
ref(*): kubernetes v1.11 support

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

**Comment by technosophos** [comment]
So we need to test this manually on:

- [ ] Kube 1.11
- [ ] Kube 1.10

And do we also need 1.9? Or is that outside of our supported versions? I can't remember how many versions of Kubernetes are considered active.

**Comment by consideRatio** [comment]
## K8s 1.11.0-gke.1 with alpha features enabled on GKE
@adamreese, using helm v2.10.0-rc.2 and your tiller image, my GKE 1.11.0-gke.1 alpha cluster failed to setup like this.

```
$ helm init --service-account tiller --tiller-image quay.io/adamreese/tiller:kube-v1.11
$HELM_HOME has been configured at /home/erik/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.

Please note: by default, Tiller is deployed with an insecure 'allow unauthenticated users' policy.
To prevent this, run `helm init` with the --tiller-tls-verify flag.
For more information on securing your installation see: https://docs.helm.sh/using_helm/#securing-your-helm-installation
Happy Helming!

$ helm version
Error: could not find a ready tiller pod

$ helm version
Error: could not find a ready tiller pod

$ helm version
Error: forwarding ports: error upgrading connection: unable to upgrade connection: error dialing backend: x509: certificate signed by unknown authority
```

## K8s 1.10.5-gke.3 with alpha features enabled on GKE
@adamreese, using helm v2.10.0-rc.2 and your tiller image, my GKE 1.10.5-gke.3 alpha cluster failed to setup like this.

```
$ helm init --service-account tiller --tiller-image quay.io/adamreese/tiller:kube-v1.11
$HELM_HOME has been configured at /home/erik/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.

Please note: by default, Tiller is deployed with an insecure 'allow unauthenticated users' policy.
To prevent this, run `helm init` with the --tiller-tls-verify flag.
For more information on securing your installation see: https://docs.helm.sh/using_helm/#securing-your-helm-installation
Happy Helming!

$ helm version
Error: could not find a ready tiller pod

$ helm version
Error: could not find a ready tiller pod

$ helm version
Client: &version.Version{SemVer:"v2.10.0-rc.2", GitCommit:"56154102a2f25ebf679c791907fd355bb0377f05", GitTreeState:"clean"}
Server: &version.Version{SemVer:"kube-v1.11+unreleased", GitCommit:"a35a66483bc36acb59473f73892caf17ef3cb3d2", GitTreeState:"clean"}
```

**Comment by technosophos** [comment]
On `K8s 1.11.0-gke.1`, there looks to be a problem with the SSL certificate. Does `kubectl` give that same error?

**Comment by consideRatio** [comment]
@technosophos I have only got the error from helm commands like `helm version` `helm test`, and `helm status` and `helm upgrade --install`. I managed to run through a guide on how to setup wordpress including a lot of kubectl and gcloud commands without issues. 

#### Fruitless investigation report
Made a diff on the output of the tiller pods yaml after loaded in the cluster, no difference except for the k8s 1.11 tiller pod had `ContainersReady   True` set in conditions.

**Comment by adamreese** [comment]
Can you create the port-forward manually?

`kubectl port-forward -n kube-system deployment/tiller-deploy 44134`

**Comment by consideRatio** [comment]
@adamreese no =/

```
> kubectl port-forward -n kube-system deployment/tiller-deploy 44134
error: error upgrading connection: unable to upgrade connection: error dialing backend: x509: certificate signed by unknown authority
```

Is this related? I don't grasp these details =/ I have...

```
gcloud config set container/new_scopes_behavior true
```

image

I could try with false, it requires recreation of the cluster... I'm on it!

**Comment by consideRatio** [comment]
The error persist after:
1. kubectl delete deploy -n kube-system tiller-deploy
2. rm -rf ~/.kube ~/.minikube ~/.helm
3. re-setup of kubectl context by running some gcloud commands...
4. helm init --service-account tiller --tiller-image quay.io/adamreese/tiller:kube-v1.11

File: pkg/kube/client.go [hunk 1]
```diff
@@ -379,17 +375,11 @@ func createResource(info *resource.Info) error {
...
-			return resource.NewHelper(info.Client, info.Mapping).Delete(info.Namespace, info.Name)
-		}
-		return err
-	}
-	c.Log("Using reaper for deleting %q", info.Name)
-	return reaper.Stop(info.Namespace, info.Name, 0, nil)
+func deleteResource(info *resource.Info) error {
+	policy := metav1.DeletePropagationBackground
+	opts := &metav1.DeleteOptions{PropagationPolicy: &policy}
+	_, err := resource.NewHelper(info.Client, info.Mapping).DeleteWithOptions(info.Namespace, info.Name, opts)
```
Comment (thomastaylor312) [inline pkg/kube/client.go]: This is super slick

File: pkg/kube/wait.go [hunk 2]
```diff
@@ -119,12 +114,12 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 				}
 				deployments = append(deployments, newDeployment)
 			case *extensions.Deployment:
-				currentDeployment, err := kcs.ExtensionsV1beta1().Deployments(value.Namespace).Get(value.Name, metav1.GetOptions{})
+				currentDeployment, err := kcs.AppsV1().Deployments(value.Namespace).Get(value.Name, metav1.GetOptions{})
```
Comment (thomastaylor312) [inline pkg/kube/wait.go]: I am almost certain that this change will break the `--wait` flag in some way, but I'll need to go back and fix this anyway

**Review by thomastaylor312 — COMMENTED** [review]
Code is looking good. I am testing this against AKS clusters to make sure it is working there as well

**Comment by thomastaylor312** [comment]
Small nit: the client version is still showing as `SemVer:"v2.8+unreleased"`. We should make sure to change that to 2.10 before we cut it

**Comment by michelleN** [comment]
@thomastaylor312 - thanks for pointing that out and yea @bacongobbler made the change from `2.8` to `2.10` in the 2.10.0-rc.3 release. 🎉

**Review by bacongobbler — APPROVED** [review]

**Review by technosophos — APPROVED** [review]

**Pull request merged**